src/lib/plugins/login.js

```diff
@@ -1,28 +1,26 @@
 /* global BigInt */
 const Vec3 = require('vec3').Vec3
 
-const path = require('path')
 const crypto = require('crypto')
-const requireIndex = require('../requireindex')
-const plugins = requireIndex(path.join(__dirname, '..', 'plugins'))
 const playerDat = require('../playerDat')
 const convertInventorySlotId = require('../convertInventorySlotId')
+const plugins = require('./index')
 
 module.exports.server = function (serv, options) {
   serv._server.on('connection', client =>
     client.on('error', error => serv.emit('clientError', client, error)))
 
   serv._server.on('login', async (client) => {
-    if (client.socket.listeners('end').length === 0) return // TODO: should be fixed properly in nmp instead
+    if (client.socket?.listeners('end').length === 0) return // TODO: should be fixed properly in nmp instead
+    if (!serv.worldsReady) throw new Error('World is still preparing.')
```

Comment:
As I remember this is another bugfix: when a player connects when world is not ready yet it crashes

Reply:
what about returning instead of throwing if this is not an error ?

Reply:
Yes of course it makes sense I fixed it here https://github.com/PrismarineJS/flying-squid/pull/633/commits/923a8296343de7daf4a31d08741b58517119c656